### PR TITLE
Fix for #3955

### PR DESCRIPTION
**Pull request by jpaas**
Ref #2955

### CODE REVIEW

File: addon/serializers/json-serializer.js [hunk 1]
```diff
@@ -716,10 +716,11 @@ export default Serializer.extend({
   */
   normalizeUsingDeclaredMapping: function(modelClass, hash) {
     var attrs = get(this, 'attrs');
-    var payloadKey, normalizedKey, key;
+    var payloadKey, key;
 
     if (attrs) {
       for (key in attrs) {
+        var normalizedKey = null;
```
Comment (wecc) [inline addon/serializers/json-serializer.js]: I think we could leave the declaration in top and just set `normalizedKey = payloadKey` in the `for`. That way we don't have to do the extra check on line 736.

**Comment by wecc** [comment]
Would be great if you could add a test for this.

**Comment by bmac** [comment]
Hey @jpaas. I'd love to get this bug fixed and this pr merged. Let me know if you need any direction with adding the test.

**Comment by jpaas** [comment]
@bmac Sorry I haven't had a chance to look at this yet, but I will try this weekend. I don't know anything about the tests for this project, so if I have any questions I'll let you know.

**Comment by fivetanley** [comment]
@jpaas LGTM, can you squash to one commit and prefix the commit message with `[BUGFIX beta]` ?

File: tests/integration/serializers/json-serializer-test.js [hunk 2]
```diff
@@ -885,3 +885,24 @@ test('normalizeResponse respects `included` items (array response)', function(as
...
+
+  var jsonHash = {
+    id: "1",
+    notInMapping: 'I should be ignored',
+    title: "Rails is omakase"
+  };
+
+  var post = env.store.serializerFor("post").normalizeResponse(env.store, Post, jsonHash, '1', 'findRecord');
+
+  assert.equal(post.data.attributes.title, "Rails is omakase");
```
Comment (pangratz) [inline tests/integration/serializers/json-serializer-test.js]: I'm just curious, but according to the test description, it looks like there should be an assertion which checks that `notInMapping` is not serialized, right?

**Comment by bmac** [comment]
LGTM

**Comment by bmac** [comment]
@jpaas Do you have time to squash this pr. Let me know if you need help with the git commands to do the squash.

File: tests/integration/serializers/json-serializer-test.js (hunk 2 above)
Comment (jpaas) [inline tests/integration/serializers/json-serializer-test.js]: Ya I suppose there should be an assert that checks it isn't added, but the primary problem this PR is trying to fix is that the value of notInMapping gets assigned to the title key.

**Comment by jpaas** [comment]
@bmac Yes please, I could use some help on what you're looking for me to do here.

**Comment by bmac** [comment]
@jpaas. 

The easiest way to squash this down would be to use the interactive rebase command. You can start it using the following command which tells git you would like to edit the last 4 commits on this branch.

`git rebase -i HEAD~4`

After you run that git should open up you editor with the following text:

```
pick a607971 ignores keys that are not found in the map                        
pick 6afdf26 after testing, need to explicitly set normalizedKey to null
pick f7f88ba implements requested changed and adds test
pick 6944873 fixes lint errors
```

We want to tell git to keep the first commit and "fixup" the 3 commits after that. "fixup" merges the commits with the previous commit and discards the commit message. We can do that by changing the word `pick` to `fixup` (or `f`).

Edit the file to look like this and save and close your editor.

```
pick a607971 ignores keys that are not found in the map                        
f 6afdf26 after testing, need to explicitly set normalizedKey to null
f f7f88ba implements requested changed and adds test
f 6944873 fixes lint errors
```

After closing the editor git will merge the last 3 commits with the first commit. You can now force push your `issue_3955` branch to this pr by running the following command: `git push -f origin issue_3955`

I hope this helps. Let me know if you run into any trouble.

**Comment by jpaas** [comment]
@bmac Thanks for the git masterclass :) Hopefully I did it right. It now looks like there are some conflicts. Not sure if that's something I need to worry about or whoever is merging the PR.

**Comment by bmac** [comment]
Thanks @jpaas. This commit was merged as 72a6290b50a5e8c01034d283ce177098ee4ceb29

**Pull request closed without merging**